pysmt/optimization/optimizer.py

```diff
@@ -87,6 +89,55 @@ def _comparation_functions(self, objective_formula):
         else:
             raise PysmtValueError("Invalid optimization function type: %s" % otype)
 
+    def _get_symbol_type(self, objective_formula):
+        otype = self.environment.stc.get_type(objective_formula)
+        if otype.is_int_type():
+            return INT
+        elif otype.is_real_type():
+            return REAL
+        elif otype.is_bv_type():
+            return BVType(otype.width)
+        else:
+            raise PysmtValueError("Invalid optimization function type: %s" % otype)
+
+    def _get_or(self, objective_formula):
```

Comment:
There is no need for this function: the logical disjunction is always `Or`, `BVOr` is the bit-wise disjunction and you do not need it.